### PR TITLE
Updated GitHub publishing with new event type, admin page + tests.

### PR DESCRIPTION
**Pull request by isoos**
- Replaces #7462
- Adds two checkboxes on the UI (screenshot below)
- Allows only config updates that enables at least one of the events.
- Verifies that the event is allowed at publishing time.

image

### CODE REVIEW

File: app/lib/frontend/templates/views/pkg/admin_page.dart [hunk 1]
```diff
@@ -283,6 +283,28 @@ d.Node _automatedPublishing(Package package) {
...
           ],
         ),
+        // events
+        d.div(
+          classes: [
+            '-pub-form-checkbox-row',
+          ],
+          child: material.checkbox(
+            id: '-pkg-admin-automated-github-push-events',
+            label: 'Enable push events',
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/frontend/templates/views/pkg/admin_page.dart:293]: `label` needs to be a `String`. Let's update this in a follow-up.

[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/frontend/templates/views/pkg/admin_page.dart]: Updated rendering:

image

PTAL.

**Review by jonasfj — APPROVED** [review]
:rocket:

**Comment by Gustl22** [comment]
Just as ref: Closes #7177

**Pull request merged**